runtime/include/chpl-format.h

```diff
@@ -28,10 +28,11 @@ extern "C" {
 #endif
 
 //
-// snprintf() a size as nnn[KMG]
+// These snprintf() a size as nnn[KMG] into the given buffer, but
+// unlike snprintf() itself, they return the buffer.
 //
-int chpl_snprintf_KMG_z(char* buf, int bufSize, size_t val);
-int chpl_snprintf_KMG_f(char* buf, int bufSize, double val);
+char* chpl_snprintf_KMG_z(char* buf, int bufSize, size_t val);
+char* chpl_snprintf_KMG_f(char* buf, int bufSize, double val);
```

Comment:
What's the motivation for returning the buffer? Just to make this cleaner to use in printf instead of having to make a call before the printf?

Reply:
Yep, that's all.  I also considered doing versions that would print into an internal TLS buffer and return that, so the caller wouldn't have to supply the buffer.  That would be even cleaner, especially for situations (in comm=ofi, not ugni) where we're using these for debug output.  (Avoids an if-test on the debug flag.)  But would have to doc that you could only have one of these "outstanding" at a time, thus for the usage in comm=ugni it wouldn't be useful.